crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -0,0 +1,383 @@
+use crate::{
+    common::{
+        fuel_core_storage_adapter::{
+            block_bytes,
+            get_block_info,
+            mint_values,
+            GasPriceSettings,
+            GasPriceSettingsProvider,
+        },
+        gas_price_algorithm::SharedGasPriceAlgo,
+        l2_block_source::FuelL2BlockSource,
+        updater_metadata::UpdaterMetadata,
+        utils::{
+            BlockInfo,
+            Error as GasPriceError,
+            Result as GasPriceResult,
+        },
+    },
+    ports::{
+        GasPriceData,
+        GasPriceServiceConfig,
+        L2Data,
+        MetadataStorage,
+    },
+    v1::{
+        algorithm::SharedV1Algorithm,
+        da_source_service::service::{
+            DaBlockCostsSource,
+            DaSourceService,
+        },
+        metadata::{
+            v1_algorithm_from_metadata,
+            V1AlgorithmConfig,
+            V1Metadata,
+        },
+        service::GasPriceServiceV1,
+    },
+};
+use anyhow::Error;
+use fuel_core_services::{
+    stream::BoxStream,
+    RunnableService,
+    ServiceRunner,
+    StateWatcher,
+};
+use fuel_core_storage::{
+    not_found,
+    transactional::AtomicView,
+};
+use fuel_core_types::{
+    fuel_tx::field::MintAmount,
+    fuel_types::BlockHeight,
+    services::block_importer::SharedImportResult,
+};
+use fuel_gas_price_algorithm::v1::AlgorithmUpdaterV1;
+
+pub struct UninitializedTask<
+    L2DataStoreView,
+    GasPriceStore,
+    Metadata,
+    DA,
+    SettingsProvider,
+> {
+    pub config: V1AlgorithmConfig,
+    pub genesis_block_height: BlockHeight,
+    pub settings: SettingsProvider,
+    pub gas_price_db: GasPriceStore,
+    pub on_chain_db: L2DataStoreView,
+    pub block_stream: BoxStream<SharedImportResult>,
+    pub(crate) shared_algo: SharedV1Algorithm,
+    pub(crate) algo_updater: AlgorithmUpdaterV1,
+    pub(crate) metadata_storage: Metadata,
+    pub(crate) da_source: DA,
+}
+
+impl<L2DataStore, L2DataStoreView, GasPriceStore, Metadata, DA, SettingsProvider>
+    UninitializedTask<L2DataStoreView, GasPriceStore, Metadata, DA, SettingsProvider>
+where
+    L2DataStore: L2Data,
+    L2DataStoreView: AtomicView<LatestView = L2DataStore>,
+    GasPriceStore: GasPriceData,
+    Metadata: MetadataStorage,
+    DA: DaBlockCostsSource,
+    SettingsProvider: GasPriceSettingsProvider,
+{
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        config: V1AlgorithmConfig,
+        genesis_block_height: BlockHeight,
+        settings: SettingsProvider,
+        block_stream: BoxStream<SharedImportResult>,
```

Comment:
Do we need to box this stream? Can't we just be generic over the stream here to be more flexible for e.g. tests?